### PR TITLE
SIL linker cleanup

### PR DESCRIPTION
**Pull request by slavapestov**

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please smoke benchmark

**Comment by slavapestov** [comment]
@swift-ci Please smoke test compiler performance

File: lib/SILOptimizer/Mandatory/MandatoryInlining.cpp [hunk 1]
```diff
@@ -663,15 +663,6 @@ class MandatoryInlining : public SILModuleTransform {
                                SetFactory, SetFactory.getEmptySet(), CHA);
     }
 
-    // Make sure that we de-serialize all transparent functions,
```
Comment (gottesmm) [inline lib/SILOptimizer/Mandatory/MandatoryInlining.cpp]: I just want to double check. What was the commit that fixed this?

File: lib/SIL/Linker.cpp [hunk 2]
```diff
@@ -56,84 +56,10 @@ bool SILLinkerVisitor::processFunction(SILFunction *F) {
...
-                                   Optional<SILLinkage> Linkage) {
-  return Loader->hasSILFunction(Name, Linkage);
-}
-
-/// Deserialize the VTable mapped to C if it exists and all SIL the VTable
-/// transitively references.
-///
-/// This method assumes that the caller made sure that no vtable existed in
-/// Mod.
-SILVTable *SILLinkerVisitor::processClassDecl(const ClassDecl *C) {
```
Comment (gottesmm) [inline lib/SIL/Linker.cpp]: Why get rid of this?

File: lib/SIL/Linker.cpp [hunk 3]
```diff
@@ -56,84 +56,10 @@ bool SILLinkerVisitor::processFunction(SILFunction *F) {
...
-  return Vtbl;
-}
-
 bool SILLinkerVisitor::linkInVTable(ClassDecl *D) {
   // Attempt to lookup the Vtbl from the SILModule.
   SILVTable *Vtbl = Mod.lookUpVTable(D);
-
-  // If the SILModule does not have the VTable, attempt to deserialize the
-  // VTable. If we fail to do that as well, bail.
-  if (!Vtbl || !(Vtbl = Loader->lookupVTable(D->getName())))
```
Comment (gottesmm) [inline lib/SIL/Linker.cpp]: And this?

**Review by gottesmm — COMMENTED** [review]
Some questions. I just want to understand the overall flow of what you are changing here.

File: lib/SILOptimizer/Mandatory/MandatoryInlining.cpp (hunk 1 above)
Comment (slavapestov) [reply lib/SILOptimizer/Mandatory/MandatoryInlining.cpp]: It wasn't a "fix" per se, just a removal of an unsound optimization. See f10f8d55dc43ad1addfc5e4854f691e0d0099049.

File: lib/SIL/Linker.cpp (hunk 2 above)
Comment (slavapestov) [reply lib/SIL/Linker.cpp]: I don't believe it was doing anything, because SILModule::lookupVTable() was already loading the vtable if one was not found.

**Comment by slavapestov** [comment]
This is obsolete

**Pull request closed without merging**